src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1135,37 +1150,32 @@ public int compare(Segment o1, Segment o2) {
 
     @Override
     public void close() throws ElasticsearchException {
-        logger.debug("close now acquire writeLock");
-        try (InternalLock _ = writeLock.acquire()) {
```

Comment:
I agree. If we fail in that case, any incoming indexing request will fail any way. IndexFailedEngineException is the right description.